Fix detection of overridden Derived Data location

PaulTaykalo opened the pull request:
There was an issue when Tuist was incorrectly resolving derived data, because of additional spaces in the `defaults` utiliy call

### Short description 📝

> Describe here the purpose of your PR.

- this fixes correctly resolving Derived Data location, if it was overridden in the Xcode

### How to test the changes locally 🧐

- Override the build location in the xcode
- Clean derived data
- Run `tuist inspect build`
- See that the error that log cannot be found will be pointed at correct location (or incorrect location without this PR)


> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/get-started) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

fortmarek reviewed (commented):
Thanks! The fix looks good 👍 Left some comments around the testing strategy.

PaulTaykalo commented:
@fortmarek A few updates were added

PaulTaykalo commented on Tests/TuistSupportTests/Utils/GitControllerTests.swift:
```diff
@@ -21,7 +21,7 @@ final class GitControllerTests: TuistUnitTestCase {
         let url = "https://some/url/to/repo.git"
         let path = try temporaryPath()
 
-        system.succeedCommand(["git", "-C \(path.pathString)", "clone \(url)"])
```
Since we've touched how succeeded Command works, we've fixed every example where it is actually used.

PaulTaykalo commented:
@fortmarek Okay, this update took a bit longer that I thought. Added test with default branch as well.

Fixed the failing tests because of more pedantica approach for the system command calls

fortmarek commented on Tests/TuistSupportTests/Utils/DeveloperEnvironmentTests.swift:
```diff
@@ -0,0 +1,72 @@
+
+    init() {
+        let mockedSystem = MockSystem()
+        system = mockedSystem
+        System._shared.mutate { $0 = mockedSystem }
```
Sorry for not catching this sooner, but rewriting the shared system means we can't run this unit tests in parallel.

It'd be great to:
- replace `System` with [Command](https://github.com/tuist/Command/). [Here's](https://github.com/tuist/tuist/blob/main/Sources/TuistAutomation/Device/DeviceController.swift#L51) an example from the Tuist codebase for how to use it.
- DI `Command` through the `init` of `DeveloperEnvironment` instead of rewriting a shared variable.

Alternatively, let's at least inject `System` through the `init` of `DeveloperEnvironment` and use the `system` variable in the `DeveloperEnvironment` methods instead of referring to `System.shared`. Something akin to:
```swift
struct DeveloperEnvironment {
  private let system: Systeming
  init(system: Systeming = System.shared) {
    self.system = sytem
  }

  func method() { system.run(...) }
}
```
